src/monitor.cc

```diff
@@ -271,67 +299,57 @@ void NodeMonitor::setStatistics() {
 
     // milliseconds
     long timeDelta = (cur_time.tv_sec * 1000 + cur_time.tv_usec / 1000)
-        - (instance_->stats_.lastTime_.tv_sec * 1000 + instance_->stats_.lastTime_.tv_usec / 1000);
+        - (stats_.lastTime_.tv_sec * 1000 + stats_.lastTime_.tv_usec / 1000);
 
     // Update the number of requests processed
     // and the ratio CPU/req.
-    instance_->stats_.lastRequests_ = currReqs;
-    instance_->stats_.lastCpuPerReq_ = (reqDelta <= 0) ? 0 : (scpu + ucpu) / reqDelta;
-    instance_->stats_.lastJiffiesPerReq_ = (reqDelta <= 0) ? 0
+    stats_.lastRequests_ = currReqs;
+    stats_.lastCpuPerReq_ = (reqDelta <= 0) ? 0 : (scpu + ucpu) / reqDelta;
+    stats_.lastJiffiesPerReq_ = (reqDelta <= 0) ? 0
         : ((float) (sticks + uticks)) / reqDelta;
 
     // Request delta - requests since last check.
-    instance_->stats_.lastReqDelta_ = reqDelta;
-    instance_->stats_.timeDelta_ = timeDelta;
+    stats_.lastReqDelta_ = reqDelta;
+    stats_.timeDelta_ = timeDelta;
 
     // Update time
-    instance_->stats_.lastTime_.tv_sec = cur_time.tv_sec;
-    instance_->stats_.lastTime_.tv_usec = cur_time.tv_usec;
+    stats_.lastTime_.tv_sec = cur_time.tv_sec;
+    stats_.lastTime_.tv_usec = cur_time.tv_usec;
 
     // Last RPS
-    instance_->stats_.lastRPS_ = (int) (reqDelta / (((double) timeDelta) / 1000));
+    stats_.lastRPS_ = (int) (reqDelta / (((double) timeDelta) / 1000));
```

Comment:
There is value in having this calculated using double temporaries, since imagine that `timeDelta` is equal to 1900, and `reqDelta` is 500.  Without the double cast/promotion, this would result in 

`(int) 500 / (1900 / 1000) = (int) 500 / 1 = 500`

Whereas with double promotion, this is

`(int) 500 / (1900 / 1000.0) = (int) 500 / 1.9 = (int) 263.157 = 263`

However, it would probably be easier to force the compiler to do this by doing what I've done in the second example here, i.e. change 1000 to 1000.0.  This will result in the entire expression being calculated using double semantics before finally being truncated using the (int) cast.